langserve/server.py

```diff
@@ -474,6 +515,11 @@ def add_routes(
             f"If specifying path please start it with a `/`"
         )
 
+    if "run_name" in config_keys:
+        raise ValueError(
+            "Cannot configure run_name. Please remove it from config_keys."
```

Comment:
Do we care about key collisions arising from identical path on 2 different API routers? (I think it's fine actually)

Reply:
oh this is a good point... we probably should?

Reply:
We derive it from `request.url.path` and always get the config based on the request?

Reply:
might limit future support for complex paths tho

Reply:
I'm OK with whatever you choose here -- just wanted to surface that collisions could potentially appear